internal/services/devcenter/dev_center_project_pool_resource.go

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package devcenter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/pools"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/projects"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/devcenter/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = DevCenterProjectPoolResource{}
+var _ sdk.ResourceWithUpdate = DevCenterProjectPoolResource{}
+
+type DevCenterProjectPoolResource struct{}
+
+func (r DevCenterProjectPoolResource) ModelObject() interface{} {
+	return &DevCenterProjectPoolResourceModel{}
+}
+
+type DevCenterProjectPoolResourceModel struct {
+	Name                               string            `tfschema:"name"`
+	Location                           string            `tfschema:"location"`
+	DevCenterProjectId                 string            `tfschema:"dev_center_project_id"`
+	DevBoxDefinitionName               string            `tfschema:"dev_box_definition_name"`
+	LocalAdministratorEnabled          bool              `tfschema:"local_administrator_enabled"`
+	NetworkConnectionName              string            `tfschema:"network_connection_name"`
+	StopOnDisconnectGracePeriodMinutes int64             `tfschema:"stop_on_disconnect_grace_period_minutes"`
+	Tags                               map[string]string `tfschema:"tags"`
+}
+
+func (r DevCenterProjectPoolResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return pools.ValidatePoolID
+}
+
+func (r DevCenterProjectPoolResource) ResourceType() string {
+	return "azurerm_dev_center_project_pool"
+}
+
+func (r DevCenterProjectPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.DevCenterProjectPoolName,
+		},
+
+		"location": commonschema.Location(),
+
+		"dev_center_project_id": commonschema.ResourceIDReferenceRequiredForceNew(&pools.ProjectId{}),
+
+		"dev_box_definition_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.DevCenterDevBoxDefinitionName,
+		},
+
+		"local_administrator_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Required: true,
+		},
+
+		"network_connection_name": {
```

Comment:
I think this would be clearer as 
```suggestion
		"dev_center_attached_network_name": {
```

Reply:
Updated